### PR TITLE
refactor(dropdownMenu): Meaganmoore/sc 183873/update menuselect component

### PR DESCRIPTION
**Pull request by mdjmoore**
## Summary

**DropdownMenu**
Shortcut ticket: [sh-183873](https://app.shortcut.com/sci-design-system/story/183873/update-menuselect-component)

DropdownMenu component already exists in Storybook, but currently is named MenuSelect. This will be an update/further specs for the current component.

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [x] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: src/core/DropdownMenu/index.tsx
Comment (tihuan) [inline src/core/DropdownMenu/index.tsx:105]:
```suggestion
                 * renderInput strips InputProps, so we explicitly pass end adornment here
```

Install https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker for spell checking if needed!

File: src/core/DropdownMenu/index.tsx
Comment (tihuan) [inline src/core/DropdownMenu/index.tsx:76]: we could probably remove this div?

File: src/core/Button/style.ts
Comment (tihuan) [inline src/core/Button/style.ts:16]:
```ts
export interface SdsProps {
  isAllCaps?: boolean;
  isRounded?: boolean;
  sdsStyle?: "minimal" | "rounded" | "square";
  sdsType?: "primary" | "secondary";
}
```

ah since `Button` component's interface doesn't include `sdsStage`, it feels like this doesn't belong here?

File: src/core/DropdownMenu/style.ts
Comment (tihuan) [inline src/core/DropdownMenu/style.ts:26]: relying on HTML structure for styling is brittle. Is there a way we can target a class name instead? Thank you!

File: src/core/MenuSelect/index.tsx
Comment (tihuan) [inline src/core/MenuSelect/index.tsx:68]: this gets called with every re-render, which generates lots of noise in the console. Can we only log it once please? 🙆‍♂️ 

e.g.,

```ts
let hasWarned = false;

export default function MenuSelect() {
  if (!hasWarned) {
    console.warn(...)
    hasWarned = true;
  }
}
```

Screen Shot 2022-05-17 at 4 43 01 PM

File: src/core/InputDropdown/style.ts
Comment (tihuan) [inline src/core/InputDropdown/style.ts:205]: since `InputDropdownProps` does include `sdsStage` and `sdsStyle`, I think it makes sense to keep these two strings here?

```ts
export interface InputDropdownProps extends CommonThemeProps {
  disabled?: boolean;
  intent?: "default" | "error" | "warning";
  label: string;
  onClick: (event: React.MouseEvent<HTMLElement>) => void;
  open?: boolean;
  sdsStage: "default" | "userInput";
  sdsStyle?: "minimal" | "square" | "rounded";
  sdsType?: "singleSelect" | "multiSelect";
  details?: string;
  counter?: string;
}
```

**Review by tihuan — COMMENTED** [review]
Looks super good! Thanks so much for the PR 🎉  ✨ 

Since this is a huge component, can we get other engineers' eyes on it too?

Thank you!

File: src/core/DropdownMenu/index.tsx
Comment (masoudmanson) [reply src/core/DropdownMenu/index.tsx:105]: Fixed 👍🏻 
Thank you for the tip.

File: src/core/DropdownMenu/index.tsx
Comment (masoudmanson) [reply src/core/DropdownMenu/index.tsx:76]: Ahh, extra <div> here :D 
Removed!

File: src/core/Button/style.ts
Comment (masoudmanson) [reply src/core/Button/style.ts:16]: Oops, well spotted :D 
Fixed 👍🏻

File: src/core/DropdownMenu/style.ts
Comment (masoudmanson) [reply src/core/DropdownMenu/style.ts:26]: Oops :D
Replaced with a proper class name 💯

File: src/core/MenuSelect/index.tsx
Comment (masoudmanson) [reply src/core/MenuSelect/index.tsx:68]: Cool,
Warnings for the `DropdownMenu` and `Button` components have been updated to only appear once. 🗡️

File: src/core/InputDropdown/style.ts
Comment (masoudmanson) [reply src/core/InputDropdown/style.ts:205]: Fixed ^_^

File: src/core/MenuItem/style.ts
Comment (mmmmmmaya) [inline src/core/MenuItem/style.ts:26]:
```suggestion
      padding: ${spacings?.s}px ${spacings?.m}px;
```

File: src/core/MenuItem/style.ts
Comment (mmmmmmaya) [inline src/core/MenuItem/style.ts:29]:
```suggestion
        font-weight: ${selected ? fontWeights?.semibold : fontWeights?.regular};
```

File: src/core/MenuItem/style.ts
Comment (mmmmmmaya) [inline src/core/MenuItem/style.ts:33]:
```suggestion
        background-color: transparent;
```

File: src/core/MenuItem/style.ts
Comment (mmmmmmaya) [inline src/core/MenuItem/style.ts:50]: can we use more specific selectors in order to use `!important` here?

File: src/core/MenuItem/style.ts
Comment (mmmmmmaya) [inline src/core/MenuItem/style.ts:59]: In general, please add semi colons to all the lines that don't have them

Comment (masoudmanson) [reply src/core/MenuItem/style.ts:59]: Ahh, the `yarn lint` removes the semicolons from the code :(

File: src/core/DropdownMenu/components/Chips/index.tsx
Comment (mmmmmmaya) [inline src/core/DropdownMenu/components/Chips/index.tsx:19]: this type assertion does not seem correct ...

File: src/core/DropdownMenu/style.ts
Comment (mmmmmmaya) [inline src/core/DropdownMenu/style.ts:122]: I'd like to avoid the use of important here too, if possible

File: src/core/MenuItem/style.ts
Comment (masoudmanson) [reply src/core/MenuItem/style.ts:26]: Fixed :D
Thanks for the tip 👍🏻

File: src/core/MenuItem/style.ts
Comment (masoudmanson) [reply src/core/MenuItem/style.ts:29]: Fixed

File: src/core/MenuItem/style.ts
Comment (masoudmanson) [reply src/core/MenuItem/style.ts:50]: Ahh, There is, of course, a better method to achieve this without using `!important` s
Nicely spotted :D

File: src/core/MenuItem/style.ts
Comment (masoudmanson) [reply src/core/MenuItem/style.ts:59]: Fixed them all :)

File: src/core/DropdownMenu/style.ts
Comment (masoudmanson) [reply src/core/DropdownMenu/style.ts:122]: `!important`s are removed :)

File: src/core/DropdownMenu/components/Chips/index.tsx
Comment (masoudmanson) [reply src/core/DropdownMenu/components/Chips/index.tsx:19]: Changed to a better assertion 👍🏻

**Review by clarsen-czi — CHANGES_REQUESTED** [review]
Looks awesome Masoud!! Just a few small tweaks and I think we should be good 💪

**Review by tihuan — APPROVED** [review]
Looks fantastic! So glad DropdownMenu replaces MenuSelect nicely for `ComplexFilter` and `Dropdown`!

Please kindly wait for Maya's feedback too before merging 🙏 

Thanks again for the amazing work 🚀 🎆 !!

**Review by clarsen-czi — CHANGES_REQUESTED** [review]
One final update on the InputDropdwn live preview and then we're good! Thanks for all the help Masoud!

**Review by clarsen-czi — APPROVED** [review]
Thanks for all the help on this one, Masoud. It looks great!!

**Review by mmmmmmaya — APPROVED** [review]

**Pull request merged**
